app/models/user.rb

```diff
@@ -48,10 +50,16 @@ def self.onboard_from_mobile(params)
     user = User.find_or_initialize_by(mobile: params[:mobile])
     user.name = params[:name]
     user.identities.find_or_initialize_by(provider: 'mobile', uid: params[:uid])
+    update_referrals(params[:referral_code]) if params[:referral_code]
     user.save_provider_auth_user
     user
   end
 
+  def update_referrals(code)
+    referral = Referral.find_by(code: code, candidate_id: nil)
```

Comment:
With the current logic an user is entitled to refer ONLY one other person, as once that happens next time this method won't find any more entry with `candidate_id: nil`. Is this the expected behaviour ?